newIDE/app/src/SceneEditor/Toolbar.js

```diff
@@ -41,6 +42,8 @@ type Props = {|
   getContextMenuZoomItems: I18nType => Array<MenuItemTemplate>,
   setZoomFactor: number => void,
   onOpenSettings?: ?() => void,
+  selectedObjectWithContext: ?ObjectWithContext,
```

Comment:
The toolbar only needs to know if an object can be renamed right now, so I think it only needs 2 props: `canRenameObject` and `onRenameObject` (this one is equivalent to your `startRenamingObject` but with the convention we try to respect in the project)

Reply:
Done!